### PR TITLE
Pallet AURA: remove pallet::getter macro and write the corresponding code

### LINKED ISSUE #3330: remove pallet::getter usage from pallet-aura
This issue implements the removal of pallet::getter in pallet-aura, as per #3326.
Discussion on the issue is here #223.

### PR DESCRIPTION
**Pull request by k-gunjan**
Removed the `pallet::getter` macro call from storage type definitions and added the corresponding implementations directly.
fixes #3330  

polkadot address: 14JzTPPUd8x8phKi8qLxHgNTnTMg6DUukCLXoWprejkaHXPz

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 3/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5231973

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5231954

File: substrate/frame/aura/src/lib.rs [hunk 1]
```diff
@@ -311,6 +309,48 @@ impl<T: Config> Pallet<T> {
...
+	///
+	/// A `BoundedVec` containing the current set of Aura authorities. The type of each authority
+	/// is defined by `T::AuthorityId`, which is associated with the specific implementation of the
+	/// Aura consensus in the pallet.
+	///
+	/// # Deprecation note
+	///
+	/// This function will be deprecated in future releases. It is recommended not to use it in new
+	/// developments.
+	pub fn authorities() -> BoundedVec<T::AuthorityId, T::MaxAuthorities> {
```
Comment (kianenigma) [inline substrate/frame/aura/src/lib.rs:330]: where are these functions used? do they even need to exist?

Comment (k-gunjan) [reply substrate/frame/aura/src/lib.rs:330]: Thank you for your review and the points raised. I appreciate the opportunity to clarify the usage and implications of the proposed changes.

I've noted that the this function is utilized not only within this pallet like [here](https://github.com/paritytech/polkadot-sdk/pull/3350/files/f92d601fbe50e45bced1ede2181dc81edd3013a1#diff-be71be33bddf8c17afa2bac0568bd30991b1098634bb3cd8a101545052849cd7R378) but also across other crates within the SDK like[ here](https://github.com/paritytech/polkadot-sdk/blob/master/cumulus/parachain-template/runtime/src/lib.rs#L549). This widespread use underscores its current necessity and potential impact on existing implementations.

In my understanding, the primary aim of this PR is to streamline the codebase by removing unnecessary macro complexity, thereby improving readability and maintainability without altering the existing functionality or interfaces. Given this scope, maintaining these function calls as-is ensures compatibility and stability across the SDK during this transitional phase. Please suggest .

If recommended I can go ahead and replace the uses here with Authorities::<T>::get() in a future iteration.

Let me know if you recommend proceeding with these adjustments now, or if we should consider them as part of a subsequent phase of optimization.

Comment (ggwpez) [reply substrate/frame/aura/src/lib.rs:330]: >If recommended I can go ahead and replace the uses here with Authorities::<T>::get() in a future iteration.

In my opinion we should do this. No need for another function.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5238135

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable-int
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5238169

File: substrate/frame/aura/src/lib.rs (hunk 1 above)
Comment (k-gunjan) [reply substrate/frame/aura/src/lib.rs:330]: Thank you for your valuable feedback. I've implemented the suggested changes in the latest commit (e81e6c4). You can view the changes[ here](https://github.com/paritytech/polkadot-sdk/pull/3350/commits/e81e6c4f86c06fd0047c6b333ff5a59e438a32c3).

Please take a look at the updated code when you have a moment. I'm open to any further suggestions or discussions if there's more to refine.

**Review by kianenigma — APPROVED** [review]

File: substrate/frame/aura/src/lib.rs (hunk 1 above)
Comment (muraca) [reply substrate/frame/aura/src/lib.rs:330]: @k-gunjan I think Oliver meant that you should not introduce these new methods in this PR.
Instead, you should go around the other crates that call the getter functions you removed, and replace them with `pallet_aura::Authorities::<T>::get()`.

**Review by muraca — CHANGES_REQUESTED** [review]

File: substrate/frame/aura/src/lib.rs (hunk 1 above)
Comment (k-gunjan) [reply substrate/frame/aura/src/lib.rs:330]: Thanks Muraca ! I'll remove definition from here and replace uses in other pallets.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5436971

**Review by s0me0ne-unkn0wn — APPROVED** [review]
LGTM! :+1:

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 3/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5440957

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 3/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5441671

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 2/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5441670

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 3/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5442642

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 2/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5447492

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 3/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5447493

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 3/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5447786

**Comment by muraca** [comment]
As a breaking change, this requires the removal of the R0 label and a prdoc

File: substrate/frame/aura/src/lib.rs [hunk 2]
```diff
@@ -369,7 +367,7 @@ impl<T: Config> FindAuthor<u32> for Pallet<T> {
 			if id == AURA_ENGINE_ID {
 				let slot = Slot::decode(&mut data).ok()?;
 				let author_index = *slot % Self::authorities_len() as u64;
-				return Some(author_index as u32)
+				return Some(author_index as u32);
```
Comment (bkchr) [inline substrate/frame/aura/src/lib.rs:370]:
```suggestion
				return Some(author_index as u32)
```

**Review by bkchr — APPROVED** [review]

**Comment by k-gunjan** [comment]
tip?

**Pull request merged**